### PR TITLE
Support for multiple extension jars by scanning the given folder

### PR DESCRIPTION
**Pull request by iNikem**
Documentation will follow later when we are ready to declare this stable.

### CODE REVIEW

File: javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/ExtensionLoader.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    List<URL> result = new ArrayList<>();
+
+    if (locationName == null) {
+      return result;
+    }
+
+    File location = new File(locationName);
+    if (location.isFile()) {
+      addFileUrl(result, location);
+    } else {
```
Comment (mateuszrzeszutek) [inline javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/ExtensionLoader.java:68]: Nit: it'll probably be a bit safer to exclude sockets, pipes and other UNIX file types:

```suggestion
    } else if (file.isDirectory()) {
```

File: javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentInstaller.java [hunk 2]
```diff
@@ -490,7 +490,7 @@ public void onComplete(
 
     private static boolean isIgnored(Class<?> c) {
       ClassLoader cl = c.getClassLoader();
-      if (cl instanceof AgentClassLoader || cl instanceof ExtensionClassLoader) {
+      if (cl instanceof AgentClassLoader) {
```
Comment (mateuszrzeszutek) [inline javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentInstaller.java:493]: What's the reason for this change? Do we want to redefine extension classes?

File: javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/ExtensionLoader.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            System.getProperty(
+                "otel.javaagent.experimental.initializer.jar",
+                System.getenv("OTEL_JAVAAGENT_EXPERIMENTAL_INITIALIZER_JAR"))));
+    // TODO when logging is configured add warning about deprecated property
+
+    List<ClassLoader> delegates = new ArrayList<>(extension.size());
+    for (URL url : extension) {
+      delegates.add(getDelegate(parent, url));
+    }
+    return new MultipleParentClassLoader(parent, delegates);
```
Comment (mateuszrzeszutek) [inline javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/ExtensionLoader.java:51]: Nit: it probably could be:
```suggestion
    return new MultipleParentClassLoader(delegates);
```
Each of the `delegates` already calls `parent`

Comment (iNikem) [reply javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/ExtensionLoader.java:51]: I actually don't know about this. We use this `MultipleParentClassLoader` to load ALL our classes, including agent's ones (this `parent` here is `AgentClassLoader`). It somehow seems strange to me to "proxy" all those load requests through the first extension classloader.

O, while writing the sentence above, I have realised that we will situation when there is NO extensions. Then we still have to have a way to delegate to `AgentClassLoader`, thus I think we need this parent here.

File: javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentInstaller.java (hunk 2 above)
Comment (iNikem) [reply javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentInstaller.java:493]: The reason for this change is there is no more `ExtensionClassLoader`. I can change for `MultipleParentClassLoader`, but there may be different instance of it, not created by us.

And I don't know if there is any actual implication of this check.

File: examples/extension/build.gradle [hunk 4]
```diff
@@ -30,6 +30,7 @@ ext {
 }
 
 repositories {
+  mavenLocal()
```
Comment (anuraaga) [inline examples/extension/build.gradle:33]: Remove or order at bottom?

**Review by anuraaga — APPROVED** [review]

File: javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentInstaller.java (hunk 2 above)
Comment (iNikem) [reply javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentInstaller.java:493]: But do we actually need this?

File: examples/extension/build.gradle (hunk 4 above)
Comment (iNikem) [reply examples/extension/build.gradle:33]: Will remove in later PR when we have suitable snapshots published.

**Review by mateuszrzeszutek — APPROVED** [review]

File: javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentInstaller.java (hunk 2 above)
Comment (trask) [reply javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentInstaller.java:493]: It's an optimization to avoid retransforming classes that were loaded by the extension before we set up the bytebuddy transformers, @laurit's idea seems simple enough to be worth keeping.

**Review by trask — APPROVED** [review]

File: javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentInstaller.java (hunk 2 above)
Comment (iNikem) [reply javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentInstaller.java]: Done.

**Review by mateuszrzeszutek — APPROVED** [review]
